packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/utils/DrawableUtils.kt

```diff
@@ -0,0 +1,27 @@
+package com.datadog.reactnative.sessionreplay.utils
+
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.InsetDrawable
+import android.graphics.drawable.LayerDrawable
+import com.facebook.react.views.view.ReactViewBackgroundDrawable
+
+internal fun getReactBackgroundFromDrawable(drawable: Drawable?): ReactViewBackgroundDrawable? {
+    if (drawable is ReactViewBackgroundDrawable) {
+        return drawable
+    }
+
+    if (drawable is InsetDrawable) {
+        return getReactBackgroundFromDrawable(drawable.drawable)
```

Comment:
If this is a risk then perhaps we need to do something for the native sdk as well, since our logic in resolveShapeStyleAndBorder also recursively drills into drawables